bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -259,6 +265,34 @@ public BigtableAsyncAdmin(CommonConnection asyncConnection) throws IOException {
     return listTables(Optional.of(pattern));
   }
 
+  /** {@inheritDoc} */
+  @Override
+  public CompletableFuture<List<TableDescriptor>> listTableDescriptors(List<TableName> tableNames) {
+    Preconditions.checkNotNull(tableNames, "tableNames is null");
+    if (tableNames.isEmpty()) {
+      return CompletableFuture.completedFuture(Collections.emptyList());
+    }
+
+    return toCompletableFuture(bigtableTableAdminClient.listTablesAsync())
+        .thenApply(
+            t ->
+                tableNames.stream()
+                    .filter(inputTableName -> t.contains(inputTableName.getNameAsString()))
+                    .map(
+                        tbName -> {
+                          try {
+                            return getDescriptor(tbName).join();
+                          } catch (CompletionException ex) {
+                            LOG.warn(
+                                "Table not found while fetching details for %s",
+                                ex, tbName.getNameAsString());
```

Comment:
I don’t think you can assume that an error means the table is not found. I think you should examine the error and check if it’s a statusruntimeexception with a not found status or a notfoundexception from gax

Reply:
As [BigtableAsyncAdmin#getDescirptor](https://github.com/googleapis/cloud-bigtable-client/blob/master/bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java#L278-L296) is handling `Status.Code.NOT_FOUND`, than re throwing exception for case other than `Status.Code.NOT_FOUND` ok?

```java
toCompletableFuture(bigtableTableAdminClient.listTablesAsync())
        .thenApply(
            t ->
                tableNames.stream()
                    .filter(inputTableName -> t.contains(inputTableName.getNameAsString()))
                    .map(
                        tbName -> {
                          try {
                            return getDescriptor(tbName).join();
                          } catch (CompletionException ex) {
                            if(ex.getCause() instanceof  TableNotFoundException){
                              LOG.warn(
                                  "Table not found while fetching details for %s",
                                  ex, tbName.getNameAsString());
                              return null;
                            }
                            throw ex;
                          }
                        })
                    .filter(Objects::nonNull)
                    .collect(Collectors.toList()));
```

Reply:
That's even better!

Reply:
Thanks for the quick response, I have updated the code. Please have a look.

Reply:
I don't think you need to log anything...no reason to panic the user. But otherwise  LGTM